hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/Route.java

```diff
@@ -38,102 +41,113 @@
-		            .log("Received response from Pharmanet")
-		            .to("log:HttpLogger?level=DEBUG&showBody=true&showHeaders=true&multiline=true")
-		            .process(new PharmaNetPayloadExtractor())
-		            
-	            //sending message to HIBC for ELIG
-	            .when(simple("${in.header.messageType} == {{hibc-r15-endpoint}} || ${in.header.messageType} == {{hibc-e45-endpoint}}"))
-	                .log("the HIBC endpoint(${in.header.messageType}) is reached and message will be dispatched to message queue(ELIG).")
-                    .setBody(simple(SampleMessages.e45ResponseMessage))
-	            
-	            //sending message to HIBC for ENROL
-	            .when(simple("${in.header.messageType} == {{hibc-r50-endpoint}}"))
-	                .log("the HIBC endpoint (${in.header.messageType}) is reached and message will be dispatched to message queue(ENROL).")
-                    .setBody(simple(SampleMessages.r50ResponseMessage))
-	            
-	            //others sending to JMB
-	            .otherwise()
-                    .log("the JMB endpoint is reached and message will be dispatched to JMB!!")
-                    .setBody(simple(SampleMessages.r03ResponseMessage))
-            .end();
-
-    }
+	private String pharmanetCert;
+
+	private static final String pharmanetCertPassword = System.getenv("PHARMANET_CERT_PASSWORD");
+
+	private static final String pharmanetUser = System.getenv("PHARMANET_USER");
+
+	private static final String pharmanetPassword = System.getenv("PHARMANET_PASSWORD");
+
+	private static ApplicationProperties properties;
+
+	private Validator validator;
+
+	@Override
+	public void configure() {
+		init();
+
+		onException(CustomHNSException.class).process(new ExceptionHandler()).handled(true);
+
+		onException(org.apache.http.conn.HttpHostConnectException.class)
+				.setHeader(Exchange.HTTP_RESPONSE_CODE, constant(500)).handled(true).setBody(constant(""))
+				.log("Failed to connect remote server");
+
+		onException(ValidationFailedException.class).log("Validation exception response: ${body}").handled(true)
+				.id("ValidationException");
+
+	
+		setupSSLConextPharmanetRegistry(getContext());
+		String pharmNetUrl = String.format(pharmanetUri
+				+ "?bridgeEndpoint=true&sslContextParameters=#%s&authMethod=Basic&authUsername=%s&authPassword=%s",
+				SSL_CONTEXT_PHARMANET, pharmanetUser, pharmanetPassword);
+		String basicToken = buildBasicToken(pharmanetUser, pharmanetPassword);
+		log.info("Using pharmNetUrl: " + pharmNetUrl);
+
+		String isFileDropsAllowed = properties.getValue(IS_FILEDDROPS_ALLOWED);
+		
+		from("jetty:http://{{hostname}}:{{port}}/{{endpoint}}").routeId("hnsecure-route")
+				.onCompletion()//It must have route scope and called only once.
+				.choice().when(header("isFileDropsAllowed").isEqualToIgnoreCase(Boolean.TRUE)).bean(ResponseFileDropGenerater.class).id("ResponseFileDropGenerater")
+				.to("log:foo").end().end()
```

Comment:
Can this live out side the `from()` block? Either towards the bottom or with the `onExceptions()`. It works here but it might just be confusing for future developers
It looks like there's some testing/debugging type logging that's still here also.

Reply:
The onCompletion method is called once per route execution. Making it global will generate two response file drops.
Tried putting it at the end of the route to avoid confusion. Doesn't work if validation fails or throws illegalargumentexception.